File: bindings/python/examples/how_tos/account_output/implicit_account_creation.py
import os

from dotenv import load_dotenv
from iota_sdk import Wallet, WalletOptions

# In this example, we create an implicit account creation address.

load_dotenv()

if 'WALLET_DB_PATH' not in os.environ:
    raise Exception(".env WALLET_DB_PATH is undefined, see .env.example")

wallet = Wallet(WalletOptions(storage_path=os.environ.get('WALLET_DB_PATH')))

# Get the implicit account address.
address = wallet.implicit_account_creation_address()
print(f'{address}')

File: bindings/nodejs/examples/how_tos/account_output/implicit-account-creation.ts
// Copyright 2024 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Wallet, initLogger } from '@iota/sdk';

// This example uses secrets in environment variables for simplicity which should not be done in production.
//
// Make sure that `example.stronghold` and `example.walletdb` already exist by
// running the `how_tos/wallet/create-wallet` example!
//
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./how_tos/account_output/implicit-account-creation.ts

// In this example we create an implicit account creation address.
async function run() {
    initLogger();
    for (const envVar of ['WALLET_DB_PATH']) {
        if (!(envVar in process.env)) {
            throw new Error(`.env ${envVar} is undefined, see .env.example`);
        }
    }

    try {
        // Create the wallet
        const wallet = await Wallet.create({
            storagePath: process.env.WALLET_DB_PATH,
        });

        // Get the implicit account address.
        const implicitAccountCreationAddress =
            await wallet.implicitAccountCreationAddress();
        console.log(
            `Implicit account creation address: ${implicitAccountCreationAddress}`,
        );
    } catch (error) {
        console.log('Error: ', error);
    }
    process.exit(0);
}

void run().then(() => process.exit());
